doc/regexp/methods.rdoc

```diff
@@ -0,0 +1,42 @@
+== \Regexp Methods
```

Comment:
I'm not sure what the purpose of having a full list of methods that accept regexp is?

Reply:
When I looked at these, I saw some methods that I've used but have not been aware that I can use a regexp (giving more power).  Just thought I'd remind others.  It's off-page, and so is pretty much free?

Reply:
i would have enjoyed such a list as a Ruby beginner, though maybe it should be made clear that it is incomplete, e.g. "These are some of the methods that accept a Regexp as an argument"

Reply:
I meant for it to be complete, and would like it to be.  Are there others?

Reply:
> i would have enjoyed such a list as a Ruby beginner, though maybe it should be made clear that it is incomplete, e.g. "These are some of the methods that accept a Regexp as an argument"

I've added a note that core methods are listed.  Are some core methods missing?

Reply:
> Are some core methods missing?

Only `String#[]=` jumps out to me. 

There might be other methods that have a more or less specific handling for regexps, either now or in the future.

Of course everything that takes an object also takes a Regexp, e.g. `Array#push`. This is another reason why I'd advocate for presenting this as a curated list.

Finally, I'm not sure `Kernel#{sub,gsub}` still exist.

Reply:
Added String#[]=;  removed Kernel.  Added a note:  is it adequate?

Reply:
I can't see a note, did you add it in the last commit?

Reply:
Should not have characterized it as a note;  I changed the language in methods.rdoc line 3, to avoid suggesting that the list is complete.